orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java

```diff
@@ -254,6 +257,20 @@ static Object readJson(String text) {
     }
   }
 
+  /**
+   * Attempts to read yaml from a text String. Will throw a parsing exception on bad yaml
+   *
+   * @param text text to read as yaml
+   * @return the object representation of the yaml text
+   */
+  static Object readYaml(String text) {
```

Comment:
ideally, these functions should live in a functionProvider - that's the new, more modular way:

Example:
https://github.com/spinnaker/orca/blob/master/orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/StageExpressionFunctionProvider.java

you also won't need to do the registration above.

Reply:
oh cool! will move to that. But how does that work with the allowUnknownKeys filtering? Do you mean only move the readYaml (and readJson/readProperties) and leave the yamlFromUrl registered on demand as this patch today?

Reply:
I see what you're saying, thanks for catching that. I guess leave it as is for now then. No point splitting up the few line function out